ui-tests/src/test/java/org/openmrs/reference/AddFormTest.java

```diff
@@ -4,37 +4,40 @@
 
 import org.junit.After;
 import org.junit.Before;
-import org.junit.Ignore;
+import org.openmrs.reference.groups.BuildTests;
+import org.junit.experimental.categories.Category;
 import org.junit.Test;
-import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
+import org.openmrs.reference.page.AdministrationPage;
 import org.openmrs.reference.page.HeaderPage;
-import org.openmrs.reference.page.HomePage;
 import org.openmrs.reference.page.ManageFormsPage;
-import org.openmrs.uitestframework.test.TestBase;
+
 
 
 /**
  * Created by nata on 24.06.15.
  */
-public class AddFormTest extends TestBase {
-    private HomePage homePage;
+public class AddFormTest extends  ReferenceApplicationTestBase {
+
     private HeaderPage headerPage;
     private ManageFormsPage manageForm;
-    private ClinicianFacingPatientDashboardPage patientDashboardPage;
+    private AdministrationPage administrationPage;
+
 
     @Before
     public void setUp() throws Exception {
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        headerPage = new HeaderPage(driver);
-        manageForm = new ManageFormsPage(driver);
-        patientDashboardPage = new ClinicianFacingPatientDashboardPage(page);
+    	 administrationPage = new AdministrationPage(page);
```

Comment:
thanks @sharif i find starting from the home page makes traversing the tests orderly ie step by step, we could actually start with ( homePage = new HomePage(page);

Reply:
Why would you start with homePage

Reply:
> Why would you start with homePage, i started with AdministrationPage because am considering adminPage as the parent page of manageForms

well starting from the home page makes the flow of tests orderly than jumping to any other page since we are simulating the user experience

Reply:
Thanks @HerbertYiga  , i dont think that every page must start from homepage to be simulated , i also think we can adopt inheritance approach which has been used from child parents, eg manageFormPage having child attributes like editForm, deleteForm,saveForm , so ideally thats the same approach that we can adopt  cc @k-joseph  any suggestion on this

Reply:
@HerbertYiga , May be to me more clear Taking a look here [homePage](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/page/HomePage.java) attributes, there is a link to manage Forms which we need  not to consider because that page is different from manageForms on[ administrationPage,](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/page/AdministrationPage.java), dont know if i make sense

Reply:
as long as the test workflow isn't broken, finding the right page and testing the right functionality is all that matters. If you can ensure the prior steps to getting to the page are reused, that would be still ok

Reply:
sure thats what was trying to achieve